src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -136,6 +140,13 @@ public void Listen(int maxIncoming = 20)
             {
                 while (!this.serverCancel.IsCancellationRequested)
                 {
+                    if (this.ConnectedNetworkPeers.Count > MaxConnectionThreshold)
+                    {
+                        this.logger.LogTrace("Maximum connection threshold [{0}] reached, stalling accepting of incoming connections.", MaxConnectionThreshold);
+                        await Task.Delay(MaxConnectionThresholdStallTime).ConfigureAwait(false);
```

Comment:
Pass the `this.serverCancel.Token` to the Task.Delay

Reply:
Fixed.